### ISSUE TITLE
Using Rigger in kotlin activities/fragments is throwing  Runtime exception

### ISSUE THREAD

**Issue by manikantagarikipati** [JustKiddingBaby/FragmentRigger#24]
java.lang.RuntimeException: Unable to start activity ComponentInfo{MyAppCompatActivity.class}: com.jkb.fragment.rigger.exception.RiggerException: UnKnown error MyAppCompatActivity@16f1f3 is not added into rigger. please check your config or contact author.

**Comment by JustKiddingBaby** [comment]
Do you turn off `Instant run`? this exception should be never throwed. please show me your code.

**Comment by manikantagarikipati** [comment]
Yes I have turned it off!.

My Activity Annotation:

```
@Puppet(containerViewId = R.id.fl_content, bondContainerView = true)
class BasePCMPinActivity : BaseActivity() {//base activity is AppCompatActivity Instance

  //here is how i'm setting fragment
 Rigger.getRigger(this).startFragment(MyFragment.Companion.newInstance());

}

@Puppet
class MyFragment:BaseFragment{// this is v4 Fragment
}
```

Also I've added all the AOP Plugins and the dependencies.

**Comment by JustKiddingBaby** [comment]
that's weird, have you tried to clean your project ? that could be useful. if not, please show me your gradle version.

**Comment by manikantagarikipati** [comment]
Yeah felt the same, I've cleaned it couple of times/ invalidate caches and all sort of playing around with it but in vain.

Here is my gradle version..
`classpath 'com.android.tools.build:gradle:3.0.1'`

And my compile and target android versions:
```
ext.androidMinSdkVersion    =  15
ext.androidSdkVersion       =  27
ext.androidToolsVersion     =  "27.0.3"
```

Let me know if you need any further information info that i can provide..

**Comment by JustKiddingBaby** [comment]
Sorry, i can't reappear this error with your gradle configuration. please give me the full gradle configuration with `project build.gradle` , `modle build.gradle` and `gradle-wrapper.properties`.

About the second error, you can try add `android.enableD8=true` In `gradle.properties` This should work. 

https://stackoverflow.com/questions/47137483/android-studio-3-0-dexarchivebuilderexception

**Comment by Yhao90** [comment]
哥们,我这也出现了,
![image](https://user-images.githubusercontent.com/13725348/35964148-ce9632bc-0cf1-11e8-9335-dcaad361d0fd.png)
![image](https://user-images.githubusercontent.com/13725348/35964172-e2beaea4-0cf1-11e8-9ddc-2058809fe103.png)

**Comment by Yhao90** [comment]
getRigger的时候rigger为空
抛出异常了就

**Comment by af913337456** [comment]
very bad lib

**Comment by Stevemoretz** [comment]
I'm having the same problem.
```
@Puppet(containerViewId = R.id.frame, stickyStack = true)
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        Rigger.getRigger(this).showFragment(HelloFragment(),R.id.frame)
```

Fragment class:
```
@Puppet
class HelloFragment : Fragment() {
```
```
        classpath "com.hujiang.aspectjx:gradle-android-plugin-aspectjx:2.0.4"
        classpath 'org.aspectj:aspectjtools:1.8.13'
apply plugin: 'android-aspectjx'
    implementation 'com.justkiddingbaby:fragment-rigger:1.4.4'
```

**Comment by Stevemoretz** [comment]
I found the problem,You need to update the library and add support for androidX.I don't know how you've written it but it shouldn't be so hard just letting androidX fragment and Activities be used.Remembering that androidX is nothing but naming convention packages.Not different code.
